src/shared/components/app/app.tsx

```diff
@@ -25,6 +26,19 @@ export class App extends Component<any, any> {
   }
 
   componentDidMount(): void {
+    const siteRes = this.isoData.site_res;
+
+    if (
+      siteRes?.site_view.site.content_warning &&
+      !(siteRes?.my_user || localStorage.getItem(adultConsentLocalStorageKey))
+    ) {
+      if (confirm(siteRes.site_view.site.content_warning)) {
+        localStorage.setItem(adultConsentLocalStorageKey, "true");
+      } else {
+        history.back();
```

Comment:
Should I include an alert like lemmy nsfw does?

Reply:
Yeah I think so. It doesn't have to be a javascript::alert like theirs, but that would probably have the best compatibility. I'd probably prefer a proper modal tho, so that we can markdown render the content warning.

The most important thing is that the site behind the alert is hidden until they click a **let me in** button.